bin/strata-cli/src/link.rs

```diff
@@ -0,0 +1,184 @@
+use std::fmt::Display;
+
+use alloy::primitives::{Address as StrataAddress, TxHash};
+use bdk_wallet::bitcoin;
+
+/// A thing that exists or is represented onchain.
+/// Currently, this is either a transaction or an address.
+pub enum OnchainObject<'a> {
+    Transaction(Txid<'a>),
+    Address(Address<'a>),
+}
+
+impl<'a> From<&'a bitcoin::Address> for OnchainObject<'a> {
+    fn from(address: &'a bitcoin::Address) -> Self {
+        OnchainObject::Address(Address::Bitcoin(address))
+    }
+}
+
+impl<'a> From<&'a StrataAddress> for OnchainObject<'a> {
+    fn from(address: &'a StrataAddress) -> Self {
+        OnchainObject::Address(Address::Strata(address))
+    }
+}
+
+impl<'a> From<&'a bitcoin::Txid> for OnchainObject<'a> {
+    fn from(txid: &'a bitcoin::Txid) -> Self {
+        OnchainObject::Transaction(Txid::Bitcoin(txid))
+    }
+}
+
+impl<'a> From<&'a TxHash> for OnchainObject<'a> {
+    fn from(txid: &'a TxHash) -> Self {
+        OnchainObject::Transaction(Txid::Strata(txid))
+    }
+}
+
+impl<'a> OnchainObject<'a> {
+    /// Create a link to the object on the given explorer.
+    /// Should be of the form `http{s}://{domain}`.
```

Comment:
```suggestion
    /// Should be of the form `http{s}://{domain}`.
    ///
    /// # Examples
    ///
    /// PLEASE ADD AN EXAMPLE.
```

Reply:
Glad that you added an example.
Because I thought reading from the docstring that you don't need the `.tld` part.
Only `https://mempool`.